### PR TITLE
New Resource: azurerm_kusto_database

### PR DESCRIPTION
**Pull request by r0bnet**
This is the second resource for this issue:
https://github.com/terraform-providers/terraform-provider-azurerm/issues/3856

It depends on another open PR ([kusto cluster](https://github.com/terraform-providers/terraform-provider-azurerm/pull/4129)).

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
@r0bnet hey would you mind rebasing this now that #4129 has been merged?

**Comment by r0bnet** [comment]
> @r0bnet hey would you mind rebasing this now that #4129 has been merged?

Hey @tombuildsstuff,
I merged master into this :)

File: azurerm/resource_arm_kusto_database.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("cluster_name", clusterName)
+
+	if location := databaseResponse.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	databaseProperties := *databaseResponse.DatabaseProperties
```
Comment (katbyte) [inline azurerm/resource_arm_kusto_database.go]: We'll need to nil check this here:
```suggestion
	if props := databaseResponse.DatabaseProperties; props != nil {
```

Comment (r0bnet) [reply azurerm/resource_arm_kusto_database.go]: Added a nil check and merged master into it.

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2019-09-04 at 10 51 45

**Comment by tombuildsstuff** [comment]
The Travis error is because of `git.apache.org` being down (because this isn't rebased off of master containing the fix) - but this is otherwise 👍

**Pull request merged**